zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -569,6 +593,45 @@ public void setConfig(Map<String, Object> config) {
     this.config = config;
   }
 
+  // apply the `interpreter-setting.json` config
+  // When creating a paragraph, it will update some of the configuration
+  // parameters of the paragraph from the web side.
+  // Need to deal with 2 situations
+  // 1. The interpreter does not have a config configuration set,
+  //    so newConfig is equal to null, Need to be processed using the
+  //    default parameters of the interpreter
+  // 2. The user manually modified the  interpreter types of this paragraph.
+  //    Need to delete the existing configuration of this paragraph,
+  //    update with the specified interpreter configuration
+  public void applyConfigSetting(Map<String, Object> newConfig) {
```

Comment:
Can we add unit test for this ? Since this is not trivial

Reply:
Added test case. in NotebookTest::testInterpreterSettingConfig()